app.js

```diff
@@ -0,0 +1,205 @@
+        try {
+            const allData = await getData('https://restcountries.com/v3.1/all?fields=name,population,currencies,region');
+
+            const totalCountries = allData.length;
+            let totalPopulation = allData.reduce((sum, c) => sum + (Number(c.population) || 0), 0);
+            const avgPopulation = Math.round(totalPopulation / totalCountries);
+
+            const regionData = {};
+            allData.forEach(country => {
+                const region = country.region || 'Unknown';
+                if (!regionData[region]) {
+                    regionData[region] = { count: 0, population: 0 };
+                }
+                regionData[region].count += 1;
+                regionData[region].population += Number(country.population) || 0;
+            });
+
+            let resultsHTML = `
+                <div><strong>Total countries result: </strong>${totalCountries.toLocaleString()}</div>
+                <div><strong>Total Countries Population: </strong>${totalPopulation.toLocaleString()}</div>
+                <div><strong>Average Population: </strong>${avgPopulation.toLocaleString()}</div>
+                <br>
+                <table>
+                    <thead>
+                        <tr>
+                            <th>Country Name</th>
+                            <th>Total Population</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+            `;
+
+            resultsHTML += allData
+                .map(country => {
+                    const countryName = country.name?.common || 'Unknown';
+                    const population = country.population ? Number(country.population) : 0;
+
+                    return `
+                        <tr>
+                            <td>${countryName}</td>
+                            <td>${population.toLocaleString()}</td>
+                        </tr>
+                    `;
+                }).join('');
+
+            resultsHTML += `
+                    </tbody>
+                </table>
+                <br>
+                <table>
+                    <thead>
+                        <tr>
+                            <th>Region</th>
+                            <th>Number of countries</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+            `;
+
+            resultsHTML += Object.entries(regionData)
+                .map(([region, data]) => `
+                    <tr>
+                        <td>${region}</td>
+                        <td>${data.count}</td>
+                    </tr>
+                `).join('');
+
+            resultsHTML += `
+                    </tbody>
+                </table>
+            `;
+
+            statsDiv.innerHTML = resultsHTML;
+        } catch (err) {
+            showError('Error fetching all countries data: ' + err.message);
+        }
+    };
+
+    const renderSearchResults = async (searchTerm) => {
```

Comment:
this function is exactly like the previous one, with very minor differences, in other words: DRY